web/src/beta/lib/core/StoryPanel/hooks.ts

```diff
@@ -61,6 +63,19 @@ export default (
     [selectedPageId, isEditable],
   );
 
+  const handlePageTime = useCallback(
+    (page: StoryPage) => {
+      const timePointField = page.property?.timePoint;
+      const currentTime = timePointField?.timePoint?.value;
+      if (!currentTime) onTimeChange?.(new Date());
+      else {
+        const getNewDate = new Date(currentTime.substring(0, 19)).getTime();
+        return onTimeChange?.(new Date(getNewDate));
+      }
+    },
+    [onTimeChange],
+  );
```

Comment:
I think the logic you have in the onTimeChange inside of the Editor should be here.
Like the flyTo, I think the StoryPanel should handle the logic without core's parent having to worry about it.